From af45b4cfc621757a2ecf966f13b7f77cbaa1df40 Mon Sep 17 00:00:00 2001
From: David Heidelberg <david@ixit.cz>
Date: Mon, 24 Aug 2026 19:29:53 +0200
Subject: [PATCH 15195/15333] dt-bindings: media: i2c: Add Samsung S5K3P9SP
 camera sensor

Document the Samsung S5K3P9SP raw image sensor, used as the front
camera of the SHIFT 6MQ (axolotl) phone.

Signed-off-by: David Heidelberg <david@ixit.cz>
---
 .../bindings/media/i2c/samsung,s5k3p9sp.yaml  | 99 +++++++++++++++++++
 1 file changed, 99 insertions(+)
 create mode 100644 Documentation/devicetree/bindings/media/i2c/samsung,s5k3p9sp.yaml

diff --git a/Documentation/devicetree/bindings/media/i2c/samsung,s5k3p9sp.yaml b/Documentation/devicetree/bindings/media/i2c/samsung,s5k3p9sp.yaml
new file mode 100644
index 000000000000..afa9d4fea85b
--- /dev/null
+++ b/Documentation/devicetree/bindings/media/i2c/samsung,s5k3p9sp.yaml
@@ -0,0 +1,99 @@
+# SPDX-License-Identifier: GPL-2.0-only
+%YAML 1.2
+---
+$id: http://devicetree.org/schemas/media/i2c/samsung,s5k3p9sp.yaml#
+$schema: http://devicetree.org/meta-schemas/core.yaml#
+
+title: Samsung S5K3P9SP Image Sensor
+
+description:
+  Samsung S5K3P9SP image sensor is a 16MP front camera sensor.
+  The widest supported output image frame size is 4608x3456 at 30 frames
+  per second, data output format is RAW10 transferred over a 4-lane MIPI
+  CSI-2 D-PHY interface.
+
+maintainers:
+  - David Heidelberg <david@ixit.cz>
+
+allOf:
+  - $ref: /schemas/media/video-interface-devices.yaml#
+
+properties:
+  compatible:
+    const: samsung,s5k3p9sp
+
+  reg:
+    maxItems: 1
+
+  clocks:
+    description: MCLK supply clock.
+    maxItems: 1
+
+  reset-gpios:
+    description: Active low GPIO connected to RESET pad of the sensor.
+    maxItems: 1
+
+  vana-supply:
+    description: Analogue voltage supply, 2.8 V.
+
+  vdig-supply:
+    description: Digital core voltage supply, 1.05 V.
+
+  vio-supply:
+    description: Digital I/O voltage supply, 1.8 V.
+
+  port:
+    $ref: /schemas/graph.yaml#/$defs/port-base
+    additionalProperties: false
+
+    properties:
+      endpoint:
+        $ref: /schemas/media/video-interfaces.yaml#
+        unevaluatedProperties: false
+
+        properties:
+          data-lanes:
+            items:
+              - const: 1
+              - const: 2
+              - const: 3
+              - const: 4
+
+        required:
+          - link-frequencies
+
+required:
+  - compatible
+  - reg
+  - port
+
+unevaluatedProperties: false
+
+examples:
+  - |
+      #include <dt-bindings/gpio/gpio.h>
+
+      i2c {
+          #address-cells = <1>;
+          #size-cells = <0>;
+
+          camera@2d {
+              compatible = "samsung,s5k3p9sp";
+              reg = <0x2d>;
+              clocks = <&camera_mclk 0>;
+              assigned-clocks = <&camera_mclk 0>;
+              assigned-clock-rates = <24000000>;
+              reset-gpios = <&gpio1 10 GPIO_ACTIVE_LOW>;
+              vana-supply = <&vreg_2p8>;
+              vdig-supply = <&vreg_1p05>;
+              vio-supply = <&vreg_1p8>;
+
+              port {
+                  endpoint {
+                      link-frequencies = /bits/ 64 <700000000>;
+                      remote-endpoint = <&mipi_csi_in>;
+                  };
+              };
+          };
+      };
+...
-- 
2.55.0

